src/lex/replace.rs

```diff
@@ -246,8 +249,8 @@ fn replace_function(
             // If this branch is matched, this is not a macro call,
             // since all other cases are covered above.
             // So just append the identifier and the current token to the stack.
-            Some(Ok(Locatable { data: _, .. })) => {
-                let token = inner.next().unwrap();
+            Some(Ok(_)) => {
+                let token = incoming.pop_front().or_else(|| inner.next()).unwrap();
```

Comment:
While this is changing, I'd suggest making a lambda for `incoming.pop_front().or_else(|| inner.next())` since it's used thrice now and it'll reduce confusion going forward.